merkletree.go

```diff
@@ -123,7 +130,7 @@ insertLoop:
 				return
```

Comment:
Need to update the `salt` here too. Maybe it'd be worth swapping in the whole `toAdd`, and only updating the `parent` and `level`, to avoid not missing any fields in the future.